src/grafana_agent.py

```diff
@@ -687,3 +771,16 @@ def _agent_version(self) -> Optional[str]:
         if result is None:
             return result
         return result.group(1)
+
+    def _update_ca(self) -> None:
+        """Updates the CA cert on disk from cert_manager."""
+        if (not self.cert.enabled) or (not self.cert.ca):
+            try:
+                self.read_file(self._ca_path)
+            except (FileNotFoundError, PathError):
+                pass
+            else:
+                self.delete_file(self._ca_path)
```

Comment:
As a quicker fix, I'm leaving this in to avoid changing both the K8S and Machine charms, since I'm not 100% sure it won't have side-effects (maybe somewhere else in the code we assume that function will fail on a missing file)